### PR TITLE
Escape control

### LINKED ISSUE #22412: {:?} should not escape printable characters
E.g. äöüß

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Add `escape_control` to char and str: Escapes C0 and C1 control chars. When printing chars or strs via `{:?}` the Debug-implementation will use those methods instead of `escape_default`. 
Fixes #22412 (?)

This is my first PR, I hope everything is ok...

### CODE REVIEW

File: src/libcore/char.rs [hunk 1]
```diff
@@ -585,4 +639,4 @@ impl Iterator for EscapeDefault {
             EscapeDefaultState::Unicode(ref mut iter) => iter.next()
         }
     }
-}
+}
```
Comment (eddyb) [inline src/libcore/char.rs]: Accidentally removed trailing newline.

File: src/libcore/char.rs [hunk 2]
```diff
@@ -394,6 +429,25 @@ impl CharExt for char {
...
+            '\t' => EscapeDefaultState::Backslash('t'),
+            '\r' => EscapeDefaultState::Backslash('r'),
+            '\n' => EscapeDefaultState::Backslash('n'),
+            '\\' => EscapeDefaultState::Backslash('\\'),
+            '\'' => EscapeDefaultState::Backslash('\''),
+            '"'  => EscapeDefaultState::Backslash('"'),
+            // Unicode-escape other C0 or C1 control character.
+            '\x00' ... '\x1f' | '\u{80}' ... '\u{9f}' => 
+                EscapeDefaultState::Unicode(self.escape_unicode()),
+            // Don't escape everything else.
```
Comment (eddyb) [inline src/libcore/char.rs]: s/everything/anything.
Also, is this such a good idea?

Comment (LukasKalbertodt) [inline src/libcore/char.rs]: Thanks. In the linked issue I proposed the idea of just escaping C0 and C1 codes. Two people in the IRC agreed...

**Comment by alexcrichton** [comment]
Thanks for the PR! At this stage of Rust, however, all new APIs require an RFC for inclusion into the standard library. New APIs also start out as `#[unstable]` after being accepted via an RFC and only later will they be maked as `#[stable]` (typically at least a release cycle).

For moving forward here this would need to open an RFC for the addition of these APIs. Feel free to let me know if you need any help writing an RFC, I'd be more than willing to review any drafts!

**Pull request closed without merging**